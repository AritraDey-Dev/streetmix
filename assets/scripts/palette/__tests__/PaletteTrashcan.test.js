/* eslint-env jest */
import React from 'react'
import { PaletteTrashcan } from '../PaletteTrashcan'
import { renderWithReduxAndIntl } from '../../../../test/helpers/render'

describe('PaletteTrashcan', () => {
  it('renders with the `-visible` class when `visible` prop is true', () => {
    const { asFragment, container } = renderWithReduxAndIntl(
      <PaletteTrashcan visible />
    )
    expect(asFragment()).toMatchSnapshot()
    expect(
      container.querySelectorAll('.palette-trashcan-visible').length
    ).toEqual(1)
  })

  it('doesnâ€™t render with the `show` class when `visible` prop is false', () => {
    const { container } = renderWithReduxAndIntl(
      <PaletteTrashcan visible={false} />
    )
    expect(
      container.querySelectorAll('.palette-trashcan-visible').length
    ).toEqual(0)
  })
})
