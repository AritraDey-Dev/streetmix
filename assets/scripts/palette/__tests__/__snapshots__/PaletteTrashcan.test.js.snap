// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteTrashcan renders with the \`-visible\` class when \`visible\` prop is true 1`] = `
<div
  className="palette-trashcan palette-trashcan-visible"
>
  <injectIntl(FormattedMessage)
    defaultMessage="Drag here to remove"
    id="palette.remove"
  />
</div>
`;
